libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.applicationinsights.core;
+
+import com.microsoft.applicationinsights.core.dependencies.http.client.protocol.HttpClientContext;
+import com.microsoft.applicationinsights.core.dependencies.http.protocol.HttpContext;
+import com.microsoft.bot.builder.BotAssert;
+import com.microsoft.bot.builder.Middleware;
+import com.microsoft.bot.builder.NextDelegate;
+import com.microsoft.bot.builder.TelemetryLoggerMiddleware;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+
+import java.util.concurrent.CompletableFuture;
+
+public class TelemetryInitializerMiddleware implements Middleware {
+    private HttpContext httpContext;
+    private TelemetryLoggerMiddleware telemetryLoggerMiddleware;
+    private Boolean logActivityTelemetry;
```

Comment:
These are `final` as in [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/TelemetryInitializerMiddleware.cs#L18) they are `readonly`